openfe/tests/setup/test_molecule.py

```diff
@@ -141,6 +141,12 @@ def test_from_sdf_string_multiple_molecules(self):
         with pytest.raises(RuntimeError, match="contains more than 1"):
             Molecule.from_sdf_string(contents)
 
+    def test_reload_hydrogens(self, serialization_template):
```

Comment:
I think I want the specific "load from file" test to exist, because that's where the actual bug is. That said, I see no harm is adding a serialization cycle test that includes the full process. Will work on that.

Bigger question is whether a lot of our test fixtures should be molecules with explicit hydrogens; serialization-related tests might be better with that. However, it would be really annoying for things like atom mappings.

(Oh, and CI seems to have awakened. That's good.)

Reply:
I see, yeah maybe we just keep the "basic tests" as they are, but if we have "real molecules" then we use explicit hydrogen mappings?

Reply:
Actually, I just noticed that the fixtures we use in `test_molecule.py` are scoped to that file only. @richardjgowers : Any opposition to changing that test suite to be entirely explicit hydrogens? The tests of network serialization in `test_network.py` would still be on implicit hydrogens (simplify the atom mapping part), but the serialization of `Molecule` would always test against explicit hydrogen.